File: Test-Case-Generation-Backend/app/models/jira.py
from pydantic import BaseModel, HttpUrl, Field
from typing import List, Optional, Dict, Any


# ==================JIRA ISSUES MODELS===============================================================
class IssueFieldsStatusCategory(BaseModel):
    self: str
    id: int
    key: str
    colorName: str
    name: str


class IssueFields(BaseModel):
    summary: str
    statusCategory: IssueFieldsStatusCategory
    description: Optional[str] = None


class JiraIssue(BaseModel):
    expand: str
    id: str
    self: HttpUrl
    key: str
    fields: IssueFields


class AllJiraIssuesResponse(BaseModel):
    issues: List[JiraIssue]
    isLast: bool

# ===============================JIRA AUTH MODELS==================================================


class JiraToken(BaseModel):
    access_token: str
    expires_at: int
    refresh_token: str
    token_type: str
    scope: str

# ===============================JIRA PROJECTS MODELS==================================================


class JiraProjectAvatarUrls(BaseModel):
    x48: HttpUrl = Field(alias="48x48")
    x24: HttpUrl = Field(alias="24x24")
    x16: HttpUrl = Field(alias="16x16")
    x32: HttpUrl = Field(alias="32x32")


class JiraProject(BaseModel):
    expand: str
    self: HttpUrl
    id: str
    key: str
    name: str
    avatarUrls: JiraProjectAvatarUrls
    projectTypeKey: str
    simplified: bool
    style: str
    isPrivate: bool
    properties: Dict[str, Any]
    entityId: Optional[str] = None
    uuid: Optional[str] = None

File: jira-api-craft/src/types/jira.ts
export type JiraProject = {
  uuid: string;
  id: string;
  key: string;
  name: string;
  avatarUrls: {
    "48x48": string;
    "24x24": string;
    "16x16": string;
    "32x32": string;
  };
  projectTypeKey: string,
    simplified: boolean,
    style: string,
    isPrivate: boolean,
    entityId: string,
};

export type JiraIssue = {
  expand:  string ,
  id :  string ,
  self :  string ,
  key :  string   ,
  fields : {
    summary :  string ,
    statusCategory : {
      self :  string ,
      id :  number,
      key :  string ,
      colorName :  string ,
      name :  string 
    },
    description :  string
  }
}
